apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -111,66 +116,63 @@ export class PackageJsonUpdater {
-      debug: debugInstall,
-      allowShrinkwrapUpdates: true,
-      bypassPolicy: false,
-      noLink: false,
-      fullUpgrade: false,
-      recheckShrinkwrap: false,
-      networkConcurrency: undefined,
-      collectLogFile: false,
-      variant: variant,
-      maxInstallAttempts: RushConstants.defaultMaxInstallAttempts,
-      pnpmFilterArguments: [],
-      checkOnly: false
-    };
-    const installManager: BaseInstallManager = InstallManagerFactory.getInstallManager(
-      this._rushConfiguration,
-      this._rushGlobalFolder,
-      purgeManager,
-      installManagerOptions
     );
 
     console.log();
-    const updatePackages: Record<string, string> = {};
-    for (const packageName of packageNames) {
-      const initialVersion: string | undefined = initialVersions.get(packageName);
-      const rangeStyle: SemVerStyle = rangeStyles.get(packageName)!;
+    const commonVersions: CommonVersionsConfiguration = this._rushConfiguration.getCommonVersions(variant);
+    const dependenciesToAddOrUpdate: Record<string, string> = {};
+    for (const { packageName, version: initialVersion, rangeStyle } of packagesToAdd) {
+      const existingSpecifiedVersions: Set<string> | undefined = allDependencyVersions.get(packageName);
+      let implicitlyPreferredVersion: string | undefined;
+      if (existingSpecifiedVersions?.size === 1) {
+        implicitlyPreferredVersion = existingSpecifiedVersions.values().next().value;
+      }
+
+      const explicitlyPreferredVersion: string | undefined =
+        commonVersions.preferredVersions.get(packageName);
+
       const version: string = await this._getNormalizedVersionSpec(
         projects,
-        installManager,
         packageName,
         initialVersion,
-        implicitlyPinned.get(packageName),
+        implicitlyPreferredVersion,
+        explicitlyPreferredVersion,
         rangeStyle
       );
-      updatePackages[packageName] = version;
+
+      dependenciesToAddOrUpdate[packageName] = version;
       console.log(colors.green(`Updating projects to use `) + packageName + '@' + colors.cyan(version));
       console.log();
+
+      if (
+        existingSpecifiedVersions &&
+        !existingSpecifiedVersions.has(version) &&
+        this._rushConfiguration.ensureConsistentVersions &&
+        !updateOtherPackages
+      ) {
+        // There are existing versions, and the version we're going to use is not one of them, and this repo
+        // requires consistent versions, and we aren't going to update other packages, so we can't proceed.
+
+        const existingVersionList: string = Array.from(existingSpecifiedVersions).join(', ');
+        throw new Error(
+          `Adding '${packageName}@${version}' ` +
+            `causes mismatched dependencies. Use the "--make-consistent" flag to update other packages to use ` +
+            `this version, or try specify one of the existing versions (${existingVersionList}).`
```

Comment:
Check that this message gets word-wrapped on the console

Reply:
It's a thrown exception, so it'll get wrapped however Rush wraps error messages by default.